README.md
# PhyloProfileData
[![Bioconductor](https://img.shields.io/badge/available%20at-Bioconductor-orange)](https://bioconductor.org/packages/PhyloProfileData)
<a href="http://bioconductor.org/checkResults/release/data-experiment-LATEST/PhyloProfileData/"><img border="0" src="http://www.bioconductor.org/shields/build/release/data-experiment/PhyloProfileData.svg" title="build results; click for full report"></a>.
[![license: MIT](https://img.shields.io/badge/license-MIT-lightgrey.svg)](https://opensource.org/licenses/MIT)


The PhyloProfileData package provides a collection of datasets to accompany the 
R package [PhyloProfile pakage](https://github.com/BIONF/PhyloProfile) 
([Tran et al. 2018](https://doi.org/10.1093/bioinformatics/bty225)), where they 
are used to illustrate how to run PhyloProfile and analyse its results. Briefly,
it contains the phylogenetic profiles, the fasta sequences and the domain 
annotations for two experimental data sets, including
1. 147 human proteins in the AMPK-TOR pathway across 83 species, and
2. 1011 BUSCO arthropoda ortholog groups across 88 species in the three domains 
of life.

# Installation

```r
if (!requireNamespace("BiocManager"))
    install.packages("BiocManager")
BiocManager::install("PhyloProfileData")
```

# Usage

The data are stored in the 
[ExperimentHub](https://bioconductor.org/packages/release/bioc/html/ExperimentHub.html) 
of [Bioconductor](http://bioconductor.org) and can be accessed using the 
following R commands:

```r
# Load the data of the PhyloProfileData package
library(ExperimentHub)
eh = ExperimentHub()
myData <- query(eh, "PhyloProfileData")
# View the metadata of this data package
myData
```

```
ExperimentHub with 6 records
# snapshotDate(): 2019-05-29 
# $dataprovider: Applied Bioinformatics Dept., Goethe University Frankfurt
# $species: NA
# $rdataclass: data.frame, AAStringSet
# additional mcols(): taxonomyid, genome, description, coordinate_1_based,
#   maintainer, rdatadateadded, preparerclass, tags, rdatapath, sourceurl,
#   sourcetype 
# retrieve records with, e.g., 'object[["EH2544"]]' 

           title                                                                         
  EH2544 | Phylogenetic profiles of human AMPK-TOR pathway                               
  EH2545 | FASTA sequences for proteins in the phylogenetic profiles of human AMPK-TOR...
  EH2546 | Domain annotations for proteins in the phylogenetic profiles of human AMPK-...
  EH2547 | Phylogenetic profiles of BUSCO arthropoda proteins                            
  EH2548 | FASTA sequences for proteins in the phylogenetic profiles of BUSCO arthropo...
  EH2549 | Domain annotations for proteins in the phylogenetic profiles of BUSCO arthr...
```


Each data set contains three files (objects) corresponding for the phylogenetic 
profiles, the FASTA sequences and the protein domain annotations. A particular 
data object can be retrieve using its ID, for example:

```r
# Retrieve FASTA sequences for proteins in the phylogenetic profiles of the 
# human AMPK-TOR pathway
ampkTorFasta <- myData[["EH2545"]]
```

For a detailed description of each data set and the belonging data objects 
please see the vignette PhyloProfileData.

```r
library(PhyloProfileData)
browseVignettes("PhyloProfileData")
```

# Bugs, Comments and Suggests
Any bug reports or comments, suggestionsare highly appreciated. Please [open an 
issue on GitHub](https://github.com/BIONF/PhyloProfileData/issues/new) or be in 
touch [via email](mailto:tran@bio.uni-frankfurt.de).

# Contributors
* [Vinh](https://github.com/trvinh)
* [Hannah](https://github.com/mueli94)

# License
This data package is released under [MIT license](https://github.com/BIONF/PhyloProfile/blob/master/LICENSE).

# How-To Cite
Ngoc-Vinh Tran, Bastian Greshake Tzovaras, Ingo Ebersberger; PhyloProfile: Dynamic visualization and exploration of multi-layered phylogenetic profiles, Bioinformatics, , bty225, https://doi.org/10.1093/bioinformatics/bty225

or use the citation function in R CMD to have it directly in BibTex or LaTeX format
```r
citation("PhyloProfileData")
```

# Contact
Vinh Tran
tran@bio.uni-frankfurt.de


NEWS
Changes in version 0.99.0 (2019-05-30)
+ Submitted to Bioconductor

\section{Version 1.6.1}{
  \itemize{
    \item Added rmarkdown, markdown to Suggests
    \item Added BiocStyle to Imports
  }
}

inst/scripts/make-data.R
library(Biostrings)

# Create AMPK-TOR phylogenetic profile data set --------------------------------
# Load data
# Source https://github.com/BIONF/phyloprofile-data/blob/master/ampk-tor.zip
tempDir <- tempdir()
temp <- tempfile(tmpdir = tempDir)
download.file(
    paste0(
        "https://github.com/BIONF/phyloprofile-data/",
        "blob/master/ampk-tor.zip?raw=true"
    ),
    temp
)

# Read AMPK-TOR phylogenetic profile
ampkTorPhyloProfile <- read.table(
    unz(temp, "ampk-tor.phyloprofile"),
    header = TRUE,
    sep = "\t",
    quote = "",
    comment.char = "",
    stringsAsFactors = FALSE
)

# Read AMPK-TOR domain annotations
ampkTorDomains <- read.table(
    unz(temp, "ampk-tor.domains_F"),
    header = FALSE,
    sep = "\t",
    quote = "",
    comment.char = "",
    stringsAsFactors = FALSE
)
colnames(ampkTorDomains) <- c("seedID", "orthoID", "feature", "start", "end")

# Read AMPK-TOR fasta sequences
unzip(temp, files = "ampk-tor.extended.fa", exdir = tempDir, overwrite = TRUE)
fastaTmp <- file.path(tempDir, "ampk-tor.extended.fa")
ampkTorFasta <- Biostrings::readAAStringSet(fastaTmp)
unlink(tempDir)

# Save into RData objects
save(ampkTorPhyloProfile, file = "ampkTorPhyloProfile.RData", compress='xz')
save(ampkTorDomains, file = "ampkTorDomains.RData", compress='xz')
save(ampkTorFasta, file = "ampkTorFasta.RData", compress='xz')

# Create BUSCO Arthropoda phylogenetic profile data set ------------------------
# Load data
# Source https://github.com/BIONF/phyloprofile-data/blob/master/arthropoda.zip
tempDir <- tempdir()
temp <- tempfile(tmpdir = tempDir)
download.file(
    paste0(
        "https://github.com/BIONF/phyloprofile-data/",
        "blob/master/arthropoda.zip?raw=true"
    ),
    temp
)

# Read Arthropoda phylogenetic profile
arthropodaPhyloProfile <- read.table(
    unz(temp, "arthropoda.phyloprofile"),
    header = TRUE,
    sep = "\t",
    quote = "",
    comment.char = "",
    stringsAsFactors = FALSE
)

# Read Arthropoda domain annotations
arthropodaDomain <- read.table(
    unz(temp, "arthropoda.domains"),
    header = FALSE,
    sep = "\t",
    quote = "",
    comment.char = "",
    stringsAsFactors = FALSE
)
colnames(arthropodaDomain) <- c(
    "seedID", "orthoID", "length", "feature", "start", "end", "weight", "path"
)

# Read Arthropoda fasta sequences
unzip(temp, files = "arthropoda.extended.fa", exdir = tempDir, overwrite = TRUE)
fastaTmp <- file.path(tempDir, "arthropoda.extended.fa")
arthropodaFasta <- Biostrings::readAAStringSet(fastaTmp)
unlink(tempDir)

# Save into RData objects
save(
    arthropodaPhyloProfile, file = "arthropodaPhyloProfile.RData", compress='xz'
)
save(arthropodaDomains, file = "arthropodaDomains.RData", compress='xz')
save(arthropodaFasta, file = "arthropodaFasta.RData", compress='xz')


inst/scripts/make-metadata.R
ampkTorProfile <- data.frame(
    Title = c(
        "Phylogenetic profiles of human AMPK-TOR pathway",
        paste0(
            "FASTA sequences for proteins in the phylogenetic profiles of ",
            "human AMPK-TOR pathway"
        ),
        paste0(
            "Domain annotations for proteins in the phylogenetic profiles of ",
            "human AMPK-TOR pathway"
        )
    ),
    Description = rep(
        paste0(
            "The phylogenetic profiles of 147 human proteins in the AMPK-TOR ",
            "pathway across 83 species in the three domains of life. This ",
            "data set contains a phylogenetic profile file, a multiple fasta ",
            "file and a protein annotation file. ",
            "Reference: ",
            "Valentin Roustan, Arpit Jain, Markus Teige, Ingo Ebersberger, ",
            "Wolfram Weckwerth, An evolutionary perspective of AMPK–TOR ",
            "signaling in the three domains of life, Journal of Experimental ",
            "Botany, Volume 67, Issue 13, June 2016, Pages 3897–3907, ",
            "https://doi.org/10.1093/jxb/erw211"
        ), 3
    ),
    BiocVersion = rep("3.10", 3),
    Genome = rep(NA, 3),
    SourceType = rep("Zip", 3),
    SourceUrl = rep(
        "https://github.com/BIONF/phyloprofile-data/blob/master/ampk-tor.zip", 3
    ),
    SourceVersion = rep("v1", 3),
    Species = rep(NA, 3),
    TaxonomyId = rep(NA, 3),
    Coordinate_1_based = rep(NA, 3),
    DataProvider = rep(
        "Applied Bioinformatics Dept., Goethe University Frankfurt", 3
    ),
    Maintainer = rep("Vinh Tran <tran@bio.uni-frankfurt.de>", 3),
    RDataClass = c("data.frame", "AAStringSet", "data.frame"),
    DispatchClass = rep("Rda", 3),
    RDataPath = c(
        "PhyloProfileData/ampkTorPhyloProfile.RData",
        "PhyloProfileData/ampkTorFasta.RData",
        "PhyloProfileData/ampkTorDomain.RData"
    ),
    Note = rep(NA, 3),
    stringsAsFactors = FALSE
)

buscoArthropodaProfile <- data.frame(
    Title = c(
        "Phylogenetic profiles of BUSCO arthropoda proteins",
        paste0(
            "FASTA sequences for proteins in the phylogenetic profiles of ",
            "BUSCO arthropoda proteins"
        ),
        paste0(
            "Domain annotations for proteins in the phylogenetic profiles of ",
            "BUSCO arthropoda proteins"
        )
    ),
    Description = rep(
        paste0(
            "The phylogenetic profiles of 1011 BUSCO arthropoda ortholog ",
            "groups across 88 species in the three domains of life. This data ",
            "set contains a phylogenetic profile file, a multiple fasta file ",
            "and a protein annotation file."
        ), 3
    ),
    BiocVersion = rep("3.10", 3),
    Genome = rep(NA, 3),
    SourceType = rep("Zip", 3),
    SourceUrl = rep(
        "https://github.com/BIONF/phyloprofile-data/blob/master/arthropoda.zip",
        3
    ),
    SourceVersion = rep("v1", 3),
    Species = rep(NA, 3),
    TaxonomyId = rep(NA, 3),
    Coordinate_1_based = rep(NA, 3),
    DataProvider = rep(
        "Applied Bioinformatics Dept., Goethe University Frankfurt", 3
    ),
    Maintainer = rep("Vinh Tran <tran@bio.uni-frankfurt.de>", 3),
    RDataClass = c("data.frame", "AAStringSet", "data.frame"),
    DispatchClass = rep("Rda", 3),
    RDataPath = c(
        "PhyloProfileData/arthropodaPhyloProfile.RData",
        "PhyloProfileData/arthropodaFasta.RData",
        "PhyloProfileData/arthropodaDomain.RData"
    ),
    Note = rep(NA, 3),
    stringsAsFactors = FALSE
)

write.csv(
    file = "inst/extdata/PhyloProfileData-metadata.csv",
    rbind(ampkTorProfile, buscoArthropodaProfile), 
    row.names=FALSE
)
